BraintreeCore/src/main/java/com/braintreepayments/api/AnalyticsBaseWorker.kt

```diff
@@ -0,0 +1,23 @@
+package com.braintreepayments.api
+
+import android.content.Context
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+
+// NEXT_MAJOR_VERSION: remove this class, it may no longer be needed
+
+/**
+ * Class for background analytics tasks.
+ * This class is used internally by the SDK and should not be used directly.
+ * It is not subject to semantic versioning and may change at any time.
+ * @suppress
+ */
+abstract class AnalyticsBaseWorker(
+    context: Context,
+    workerParams: WorkerParameters
+) : Worker(context, workerParams) {
+
+    internal open fun createAnalyticsClientFromInputData(): AnalyticsClient {
+        return AnalyticsClient(applicationContext)
```

Comment:
Kotlin newb question - why don't we need to do `context.applicationContext` to access `applicationContext`?

Reply:
It's a WorkManager base class that has an [applicationContext](https://developer.android.com/reference/androidx/work/ListenableWorker#getApplicationContext()) property. I agree it is confusing here with context being highly visible in the constructor.

I see in one of the [JetPack examples](https://developer.android.com/guide/background/persistent/getting-started/define-work#input_output) they name the constructor parameter `appContext`, which is interesting. My guess is the `applicationContext` property exists for defensive programming to discourage leaking an `Activity` context, since calling `context.applicationContext` on an application context should return a reference to itself.

^ I wish I didn't have to write that last sentence but this is Android lol.

Reply:
Ah interesting, thank you!